tasking/pulp/tasking/base.py

```diff
@@ -141,16 +142,18 @@ def delete_worker(name, normal_shutdown=False):
         msg = msg % {'name': name}
         _logger.error(msg)
 
-    worker = Worker.objects.get(name=name)
-
-    # Cancel all of the tasks that were assigned to this worker's queue
-    for task_status in worker.tasks.filter(state__in=TASK_INCOMPLETE_STATES):
-        cancel(task_status.pk)
+    try:
+        worker = Worker.objects.get(name=name)
+    except Worker.DoesNotExist:
+        pass
+    else:
+        # Cancel all of the tasks that were assigned to this worker's queue
+        for task_status in worker.tasks.filter(state__in=TASK_INCOMPLETE_STATES):
+            cancel(task_status.pk)
 
-    worker.delete()
+        worker.delete()
 
-    if name.startswith(TASKING_CONSTANTS.RESOURCE_MANAGER_WORKER_NAME):
-        TaskLock.objects.filter(name=name).delete()
+    TaskLock.objects.filter(name=name).delete()
```

Comment:
Sure, I will update it. It is just want I recalled from our discussion but I could easily mixed it up.
And I agree it is more explicit and guarded way to do it.